docs/dynamic-bundling/index.md

```diff
@@ -40,3 +40,130 @@ Most others (YUI, `ResourceLoader`) focus more on pure concatenation without exp
 [ResourceLoader-blog]: https://phabricator.wikimedia.org/phame/live/7/post/175/wikipedia_s_javascript_initialisation_on_a_budget/
 [YUI-ComboHandler]: https://yuilibrary.com/yui/docs/yui/create.html#using-a-combohandler
 [lawnsea]: https://twitter.com/lawnsea
+
+## Use Cases
+
+### Chunk Batching
+
+*Applies to: `Codex`, `MSS`.*
+
+This use case could also be called "fine-grained chunk splitting" since these
+two goals are tightly related. Compression and other protocol-level concerns
+are fundamentally at odds with a desire to ship very small units of code
+("chunks" of code). The central question here is: Can we mitigate this effect
+and still retain the same level of chunk splitting?
+
+1. Send only the code that is requested.
+   Counter-Example: The traditional webpack commons chunks contains all
+                    3rd-party code even if only one library is used on the
+                    current page.
+2. Send code only once.
+   Counter-Example: Without chunk splitting, a module used in more than one
+                    entrypoint will be downloaded and executed multiple times.
+3. Any number of chunks can be downloaded using a single request.
+   Counter-Example: A script tag with a static file server will only download
+                    exactly one chunk per request.
+
+#### Chunk Batching API
+
+Both Codex and MSS are implementing chunk batching by combining a web service
+to serve concatenated chunks with a client-side runtime to manage chunk loads.
+These two chunk batching solutions independently added similar features for
+the API:
+
+* Specify a set of requested chunks.
+* Include all dependencies of the requested chunks as an option.
+* Exclude a set of already loaded chunks.
+* Pass a compressed representation of feature flags to customize the response.
+* A combined version for the overall chunk set and no versioning of individual
+  chunks.
+
+Some of these choices (e.g. excludes, combined version) are a based on the
+observation that giving the client full knowledge of the chunk graph gets less
+practical as the size of the graph expands. It is further complicated by
+customized assets where chunk dependencies may vary between different
+variants/users. See: The client cache manifest problem.
+
+### Customized Assets
+
+*Applies to: `Codex`, `MSS`.*
+
+In the presence of user preferences, themes, feature flags, and experiments,
+it can be impractical to render every possible combination of minor variations
+in the assets ahead of time. The easy solution is to include all variations
+regardless of how applicable they are to the current user. Runtime conditionals
+are then used to activate the parts of the returned code that is actually
+needed. This is wasteful in regards to network transfer and parse/execute times
+and it also risks exposing experimental features.
```

Comment:
Should we also mention the solution that deploy all variations?

Reply:
As in: Add more details beyond "it's impractical" to this?

> it can be impractical to render every possible combination of minor variations
> in the assets ahead of time.

Or are you referring to  a  different solution?

Reply:
Added  some more meat to the "deploy all variations" ("prebuild all variations") section.

Reply:
Yeah that LG :)